x-pack/plugins/stack_connectors/server/connector_types/gemini/gemini.ts

```diff
@@ -0,0 +1,259 @@
+      logger: this.logger,
+      savedObjectsClient: this.savedObjectsClient,
+      dashboardId,
+      genAIProvider: 'Gemini',
+    });
+
+    return { available: response.success };
+  }
+
+  private async runApiLatest(
+    params: SubActionRequestParams<RunApiLatestResponse> 
+  ): Promise<RunActionResponse> {
+
+    const response = await this.request(params);
+    const candidate = response.data.candidates[0];
+    const completionText = candidate.content.parts[0].text;
+    return { completion: completionText }
+
+  }
+
+  /**
+   * This method is called on TEST subaction
+   * responsible for making a POST request to the external API endpoint and returning the response data
+   * @param body The stringified request body to be sent in the POST request.
+   * @param model Optional model to be used for the API request. If not provided, the default model from the connector will be used.
+   */
+   public async runTestApi({ body, model: reqModel }: RunActionParams): Promise<RunActionResponse> {
+    // set model on per request basis
+    const currentModel = reqModel ?? this.model;
+    const path = `/v1/projects/${this.gcpProjectID}/locations/${this.gcpRegion}/publishers/google/models/${currentModel}:generateContent`;
+    const accessToken = this.secrets.accessToken;
+    const data = JSON.stringify(JSON.parse(body)['messages']);
+
+    const requestArgs = {
+      url: `${this.url}${path}`,
+      method: 'post' as Method,
+      data: data,
+      headers: { 
+        'Authorization': `Bearer ${accessToken}`,
+        'Content-Type': 'application/json'
+      },
+      timeout: 120000,
+    };
+    
+    return this.runApiLatest({ ...requestArgs, responseSchema: RunApiLatestResponseSchema });
+  }
+
+  /**
+   * responsible for making a POST request to the Vertex AI API endpoint and returning the response data
+   * @param body The stringified request body to be sent in the POST request.
+   * @param model Optional model to be used for the API request. If not provided, the default model from the connector will be used.
+   */
+  public async runApi({
+    body,
+    model: reqModel,
+    signal,
+    timeout,
+   }: RunActionParams): Promise<RunActionResponse> {
+   // set model on per request basis
+   const currentModel = reqModel ?? this.model;
+   const path = `/v1/projects/${this.gcpProjectID}/locations/${this.gcpRegion}/publishers/google/models/${currentModel}:generateContent`;
+   const accessToken = this.secrets.accessToken;
+   const data = JSON.stringify(JSON.parse(body)['messages']);
+   const payload = formatGeminiPayload(data);
+   
+   const requestArgs = {
+     url: `${this.url}${path}`,
+     method: 'post' as Method,
+     data: JSON.stringify(payload),
+     headers: {
+       'Authorization': `Bearer ${accessToken}`,
+       'Content-Type': 'application/json'
+     },
+     signal,
+     timeout: 120000,
+   };
+
+   return this.runApiLatest({ ...requestArgs, responseSchema: RunApiLatestResponseSchema });
```

Comment:
This seems to be leftover copy/paste following the Bedrock connector. Bedrock has 2 api schemas, and therefore we have 2 run methods `runApiLatest` and `runApiDeprecated`. Since there is only one run method here, we should keep it contained within `runApi`

Reply:
done